httpbp/middlewares.go

```diff
@@ -246,3 +247,35 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+// recoverPanic recovers from any panics, logs them, and sets the returned error
+// to a generic 500 error. recoverPanic is always the last middleware in the
+// middleware chain, so it is the first one when returning which lets the error
+// bubble up into other middlewares. Since it is always added to the middleware
+// chain is a specific position, it is not exported.
+func recoverPanic(name string, next HandlerFunc) HandlerFunc {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				var rErr error
+				if asErr, ok := r.(error); ok {
+					rErr = asErr
+				} else {
+					rErr = fmt.Errorf("panic in %q: %+v", name, r)
+				}
+				log.C(ctx).Errorw(
```

Comment:
just to triple check, the default logger logs call stacks for errors right?

Reply:
yes, it adds stack trace on error and above levels.